go-controller/pkg/allocator/udn/ips.go

```diff
@@ -0,0 +1,56 @@
+package udn
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+	utilnet "k8s.io/utils/net"
+)
+
+const (
+	masqueradeIPv4IDName = "v4-masquerade-ips"
+	masqueradeIPv6IDName = "v6-masquerade-ips"
+)
+
+type MasqueradeIPs struct {
+	Shared net.IP
+	Local  net.IP
+}
+
+// AllocateV4MasqueradeIPs will return local and shared gateway masquerade IPv4 addresses calculated from
+// the networkID argument
+func AllocateV4MasqueradeIPs(networkID int) (*MasqueradeIPs, error) {
+	return allocateMasqueradeIPs(masqueradeIPv4IDName, config.Gateway.V4MasqueradeSubnet, networkID)
+}
+
+// AllocateV4MasqueradeIPs will return local and shared gateway masquerade IPv6 addresses calculated from
+// the networkID argument
+func AllocateV6MasqueradeIPs(networkID int) (*MasqueradeIPs, error) {
+	return allocateMasqueradeIPs(masqueradeIPv6IDName, config.Gateway.V6MasqueradeSubnet, networkID)
+}
+
+func allocateMasqueradeIPs(idName string, masqueradeSubnet string, networkID int) (*MasqueradeIPs, error) {
+	ip, netCidr, err := net.ParseCIDR(masqueradeSubnet)
+	if err != nil {
+		return nil, err
+	}
+	ipOffset := types.UserDefinedNetworkMasqueradeIPBase + (networkID-1)*2
```

Comment:
network ID cannot be 0 right? should we guard against this happening and/or document it in the code here?
also, document, why multiple by 2

Reply:
I am going to replace the for loop with a pair of calls so we can remove a little more math.

Reply:
Added a check and explanation of why we multiply by 2